### ISSUE TITLE
Empty or white space strings equal 0

### ISSUE THREAD

**Issue by griffpatch** [LLK/scratch-vm#435]
### Expected Behavior

("   " = 0) should resolve to false.

### Actual Behavior

It resolves to true.

### Steps to Reproduce

Add an equality comparison block to an empty project, enter a 0 on one side and leave the other black. Click it to get the answer 'true'. Try entering an few white spaces on the empty side and repeat... true again. This is because Javascript counts white space as a number. If it helps, here is one possible fix:

```
Cast.compare = function (v1, v2) {
    var n1 = Number(v1);
    var n2 = Number(v2);

    // Check for empty or white space strings if number is 0

    if (n1 === 0 && (v1 == null || typeof v1 === 'string' && v1.trim().length === 0)) {
        n1 = NaN;
    }
    if (n2 === 0 && (v2 == null || typeof v2 === 'string' && v2.trim().length === 0)) {
        n2 = NaN;
    }

    // Now continue as normal

    if (isNaN(n1) || isNaN(n2)) {
        // At least one argument can't be converted to a number.
        // Scratch compares strings as case insensitive.
        var s1 = String(v1).toLowerCase();
        var s2 = String(v2).toLowerCase();
        return s1.localeCompare(s2);
    } else {
        // Compare as numbers.
        return n1 - n2;
    }
};

```

I can create a patch, but my branch is pretty out of sync with your main branch due to my optimisation tests and it will take me a little time to get it in order.

### Operating System and Browser

Any

**Comment by towerofnix** [comment]
I believe this is fixed?

Test project ID: [150822401](https://scratch.mit.edu/projects/150822401/)

Scratch 2.0:

![Scratch 2.0](https://cloud.githubusercontent.com/assets/9948030/24045065/902b289a-0afc-11e7-9014-75e589658fad.png)

Scratch 3.0, on [an old version](https://github.com/LLK/scratch-vm/tree/635de4113d652fa321c45eb04bf6056d6309da86):

![Old Scratch 3.0](https://cloud.githubusercontent.com/assets/9948030/24045038/81976f64-0afc-11e7-8184-2240e9e1a5d0.png)

Scratch 3.0, the [current version](https://llk.github.io/scratch-gui/#150822401):

![Present Scratch 3.0](https://cloud.githubusercontent.com/assets/9948030/24045095/a9b06988-0afc-11e7-8eee-24ab600a429f.png)

<sup>(edit: grammar)</sup>

**Comment by thisandagain** [comment]
Resolved in #438
